auxx/src/Mode.hs

```diff
@@ -79,6 +81,43 @@ data CmdCtx = CmdCtx
     }
 
 type AuxxMode = ReaderT AuxxContext Production
+type MonadAuxxMode m =
+    ( HasConfigurations
+    , MonadReader AuxxContext m
+    , MonadMockable m
+    , MonadSlots AuxxContext m
+    , HasLoggerName m
+    , CanJsonLog m
+    , MonadDBRead m
+    , MonadDB m
+    , MonadBlockDBGenericWrite BlockHeader Block Undo m
+    , MonadBlockDBGeneric BlockHeader Block Undo m
+    , MonadBlockDBGeneric (Some IsHeader) SscBlock () m
+    , MonadGState m
+    , MonadBListener m
+    , MonadBalances m
+    , MonadTxHistory m
+    , MonadKnownPeers m
+    , MonadFormatPeers m
+    , MonadAddresses m
+    , MonadKeysRead m
+    , MonadKeys m
+    , MonadTxpLocal m
+    , MonadTxpLocal (BlockGenMode EmptyMempoolExt m)
+    , MonadMask m
+    , CanLog m
+    , AddrDataIsPublicKey m
+    , Eq (Promise m (Maybe ())) -- what?
+    , MonadRandom m
+    , Default (MempoolExt m)
+    , MonadTxpLocal (BlockGenMode (MempoolExt m) m)
+    , MonadFail m
+    )
+
+-- For MonadAuxxMode we need to make sure that AddrData m in MonadAdresses
+-- is indeed PublicKey, as it is for AuxxMode.
+class (AddrData m ~ PublicKey) => AddrDataIsPublicKey m
```

Comment:
I don't understand why this class is defined. Can't you just use `AddrData m ~ PublicKey` above?

Besides, seeing that you have only a single instance here, `instance AddrDataIsPublicKey`, it made me realize that you could do this:

```
type MonadAuxxMode m = m ~ AuxxMode
```